CoreLayersView/src/au/gov/asd/tac/constellation/views/layers/LayersViewTopComponent.java

```diff
@@ -76,12 +81,25 @@ protected LayersViewPane createContent() {
 
     @Override
     protected void handleNewGraph(final Graph graph) {
-        // Old code - implement new handler if needed
-//        if (attributeCalculatorPane != null && graph != null) {
-//            attributeCalculatorPane.updateAttributes(graph);
-//        }
+        if (graph != null) {
+            layersViewPane.setDefaultLayers();
+            layersViewController.readState();
+        }
+    }
+    
+    @Override
+    protected void handleGraphOpened(final Graph graph) {
+        if (graph != null) {
+            layersViewPane.setDefaultLayers();
+            layersViewController.readState();
+        }
+    }
+    
+    @Override
+    protected void handleComponentOpened() {
+        layersViewPane.setDefaultLayers();
+        layersViewController.readState();
```

Comment:
I note that this function as well as the two above it (handleNewGraph, handleGraphOpened) have very similar implementations. Maybe this is something that can be taken advantage of.